### ISSUE TITLE
CI issues due to new {text shaping} dependency

### ISSUE THREAD

**Issue by pat-s** [r-lib/ragg#57]
Hi @thomasp85,

the new {ragg} release requires {textshaping} which again relies on system libs `harfbuzz` and `fribidi`.
Since {pkgdown} pull in {ragg}, the lack of the mentioned system libs breaks a lot of CI workflows ([example](https://github.com/mlr-org/mlr3spatiotempcv/runs/1212844810?check_suite_focus=true#step:12:2502))

I can adjust for this upstream in {tic} but many other users might eventually face issues as well. 
Can you let me know if this requirement will persist to prevent unneeded work? Thanks :)

**Comment by thomasp85** [comment]
Thanks - I'm aware and in the process of getting the libs picked up automatically

**Comment by pat-s** [comment]
Thanks. @thomasp85 Sorry for putting pressure on this but this issue currently breaks *all* of our CI builds (multiple orgs) due to {pkgdown} and deps being installed from source.

Do you plan to release a hotfix for this in the next days? If not, I need to update all YMLs in more than 50 repos to get builds working again.

> getting the libs picked up automatically

Do you mean that you wanna ship them within the package source or is it more about the linking against local installs? If it is the latter, than I need to take action anyhow (i.e. install the libs in the first place).

**Comment by thomasp85** [comment]
Looking through your CI script I can see that you do not have any steps for automatically detecting and installing system requirements - I would highly advice that you do this, because there's nothing else I can do than making sure these are getting detected correctly. I'd advice you to follow the same approach as https://github.com/r-lib/actions to avoid headaches in the future - specifically having something like:

```
- name: Install system dependencies
        if: runner.os == 'Linux'
        run: |
          while read -r cmd
          do
            eval sudo $cmd
          done < <(Rscript -e 'writeLines(remotes::system_requirements("ubuntu", "20.04"))')
```

in your script.

**Comment by pat-s** [comment]
Thanks for pointing this out. I haven't used it until now because it only worked on Linux. However, since our builds are using binaries on macOS now, only Linux is left for source installs.

I really like this approach. However, ACAICS it currently only looks into the `SystemRequirements` field of the checked package (and only a path can be given, no single package name) and recursive system libs reps (such as the ones from {ragg} here) are not picked up.
As a workaround I'll include the {ragg} deps into the `SystemRequirements` field of {tic} so it gets installed in every build. This is somewhat of an overhead to builds not requiring {pkgdown} in their builds but maybe the heuristics becomes smarter in the near future.

I see that you added the rules for harfbuzz and fribidi recently. When running `remotes::system_requirements()` locally in a repo with those libs specified, they do not get picked up. How can I tell `remotes::system_requirements()` to use the latest rules?

**Comment by thomasp85** [comment]
Is there any reason why you are building pkgdown on Linux and not macOS?

You are correct about the lookup - I forgot that pkgdown was not a dependency of your project... The best way forward on Linux is probably to install them manually before installing pkgdown. I can see how this is suboptimal but we'll be looking into a better way. In general I would use macOS for this as it doesn't have to deal with system requirements at all when installing pre-compiled packages

**Comment by pat-s** [comment]
> Is there any reason why you are building pkgdown on Linux and not macOS?

The default {tic} templates build the pkgdown site on macOS.
In the mlr-org we recently switched to Linux only unless we have reasons for a os-dependent behaviour of the package (parallel stuff, etc).
Hence, in all these repos the pkgdown site is build on Linux and fails now.

I would prefer a generic fix which prevents me from having to update all the repos again in a short time. So the generic "auto-detect system libs" approach would be my fav but does not work yet so I I guess I might go for a manual lib installation for these specific linux builds.

I've heard about the generic `<org>/.github` repo lately which might come in handy here.

Closing here since I think there is nothing to be solved on your side. Thanks for the quick and meaningful replies 👍
